alacritty/src/macos/proc.rs

```diff
@@ -0,0 +1,117 @@
+use std::ffi::{CStr, CString};
+use std::io;
+use std::mem::{self, MaybeUninit};
+use std::os::raw;
+use std::path::PathBuf;
+
+pub fn cwd(pid: raw::c_int) -> io::Result<PathBuf> {
```

Comment:
I've been considering this for a while and I think this definitely is something that should be discussed:

The return value of `io::Result` seems kinda misleading here. We have multiple places where we us errors that are clearly not io errors and turn them into `io::Errorkind::Other` just so we can return them here.

The only reason for doing this as far as I can tell would be laziness, since it's just "easier" to use `io::Error` for everything rather than providing separate errors depending on failure case. I think we should probably have higher standards than this.

Either we want to offer good errors and define a custom error enum that implements `From<io::Error>`, or we just say the error doesn't matter at all and we return an `Option<PathBuf>`. But the "excessive" use of `io::ErrorKind::Other` here is kinda rubbing me the wrong way.

Reply:
We return last_os_error in some cases though, so the usage of os error seems legit?

Reply:
Always returning `io::Error` is only legit if all your errors are IO errors. Some errors being io errors doesn't justify converting all your errors to io errors.

Reply:
Well, most of your errors are IO errors, and you have one case when you decided to fallback. Converting originally IO errors to some other errors sounds stranger than using `io::Error`.

Reply:
Though, I can remove one io error that doesn't make sense.

Reply:
You're not converting IO errors to anything. This is the same standard error handling we do everywhere else:

https://github.com/alacritty/alacritty/blob/539855cdb92c44dc0f755e6ea6debcf13a77f049/alacritty/src/config/mod.rs#L38-L50